src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -541,6 +558,28 @@ object AppDefinition {
       (cmd ^ args) || (!(cmd || args) && container)
     }
 
+  private val complyWithSchedulerLabelRules: Validator[AppDefinition] =
+    isTrue("Scheduler AppDefinition must define required labels") { app =>
+      val isNormalApp = !app.isScheduler
+      val containsRequiredSchedulerLabels =
+        app.labels.get(AppDefinition.Labels.DcosPackageFrameworkName).exists(_.nonEmpty) &&
+          app.labels.get(AppDefinition.Labels.DcosMigrationApiVersion).contains("v1") &&
+          app.labels.get(AppDefinition.Labels.DcosMigrationApiPath).exists(_.nonEmpty) &&
+          app.labels.get(AppDefinition.Labels.SingleInstanceApp).contains("true")
+
+      isNormalApp || containsRequiredSchedulerLabels
+    }
+
+  private val complyWithSingleInstanceLabelRules: Validator[AppDefinition] =
+    isTrue("Single instance app may only have one instance") { app =>
+      (!app.isSingleInstance) || (app.instances <= 1)
+    }
+
+  private val complyWithReadinessCheckRules: Validator[AppDefinition] =
+    isTrue("Only Scheduler AppDefinitions may define at most one readinessCheck") { app =>
```

Comment:
Ok. I moved the readiness validation to a dedicated test suite because the `AppDefinitionValidatorTest` has already 500 LOC and covers various vaidation facets.